### PR TITLE
Allow @ in tracestate key names for multi-tenant vendors

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
#137

### CODE REVIEW

File: trace_context/HTTP_HEADER_FORMAT_RATIONALE.md [hunk 1]
```diff
@@ -64,6 +64,11 @@ Lowercase names has a few benefits:
 Url encoding is low-overhead way to encode unicode characters for non-latin characters in the 
 values. Url encoding keeps a single words in latin unchanged and easy readable.
 
+## Vendor name in a key
+
+Sign `@` is allowed in a key for easy parsing of vendor name out of tracestate key. Idea is that with the registry of trace vendors one can easily understand the vendor name and how to parse it's trace state. Without `@` sign parsing will be more complicated. Also `@` sign has known semantics in
+addressing for protocols like ftp and e-mails.
```
Comment (discostu105) [inline trace_context/HTTP_HEADER_FORMAT_RATIONALE.md:70]: unintentional line-break?

Comment (discostu105) [inline trace_context/HTTP_HEADER_FORMAT_RATIONALE.md:70]: Also, not sure if "Also `@` sign has known semantics in addressing for protocols like ftp and e-mails" really adds a lot here. Would it hurt to remove this?

Comment (SergeyKanzhelev) [reply trace_context/HTTP_HEADER_FORMAT_RATIONALE.md:70]: it's markdown

Comment (SergeyKanzhelev) [reply trace_context/HTTP_HEADER_FORMAT_RATIONALE.md:70]: I think it's fine. Any way to explain it easier?

Comment (discostu105) [reply trace_context/HTTP_HEADER_FORMAT_RATIONALE.md:70]: I't ok, I'm also fine with keeping this.

**Review by AloisReitbauer — APPROVED** [review]

File: trace_context/HTTP_HEADER_FORMAT.md [hunk 2]
```diff
@@ -225,10 +225,11 @@ Identifiers are short (up to 256 characters) textual identifiers.
 
 ```
 key = lcalpha 0*255( lcalpha / DIGIT / "_" / "-"/ "*" / "/" )
+key = lcalpha 0*240( lcalpha / DIGIT / "_" / "-"/ "*" / "/" ) "@" lcalpha 0*13( lcalpha / DIGIT / "_" / "-"/ "*" / "/" )
 lcalpha    = %x61-7A ; a-z
 ```
 
-Note that identifiers MUST begin with a lowercase letter, and can only contain lowercase letters `a`-`z`, digits `0`-`9`, underscores `_`, dashes `-`, asterisks `*`, and forward slashes `/`.
+Note that identifiers MUST begin with a lowercase letter, and can only contain lowercase letters `a`-`z`, digits `0`-`9`, underscores `_`, dashes `-`, asterisks `*`, and forward slashes `/`. For multi-tenant vendors scenarios `@` sign can be used to prefix vendor name. Suggested use is to allow set tenant id in the beginning of key like `fw529a3039@dt` - `fw529a3039` is a tenant id and `@dt` is a vendor name. Searching for `@dt=` would be more robust for parsing (searching for all vendor's keys).
```
Comment (codefromthecrypt) [inline trace_context/HTTP_HEADER_FORMAT.md:232]: I think you all are way too quick to make decisions like this.

**Comment by codefromthecrypt** [comment]
for posterity there was mention about no objections. in the doc it also
said that many people were only partially available monday. can folks that
agree we should document this practice please make yourselves known, or at
least say who was present when no objections were claimed?

it is good to make decisions but it should be transparent which
"implementors" will be using this and how. I am not saying this is a bad
call just didn't see folks outside DT talking about their use cases for
this.

**Pull request merged**